packages/wxt/src/storage.ts

```diff
@@ -158,7 +172,27 @@ function createStorage(): WxtStorage {
           });
         }),
       );
-      return results.flat();
+
+      const flatResults = results.flat();
+
+      if (itemMap.size > 0) {
+        // If WxtStorageItems were passed, return an object
+        const returnObj: Record<string, any> = {};
+        flatResults.forEach(({ key, value }) => {
+          const item = itemMap.get(key);
+          if (item) {
+            const itemKey = keys.find((k) => k === item) as WxtStorageItem<
+              any,
+              any
+            >;
+            returnObj[itemKey.storageKey] = value;
+          }
+        });
+        return returnObj;
+      } else {
+        // Otherwise, return the array of results (original behavior)
+        return flatResults;
+      }
```

Comment:
I think regardless of the input types, we should return the same thing here. What if we need to add support for mixing keys and items? Should we support that right now? Probably.

Reply:
Looking back at it makes more sense to do it as an array yeah, since you can just get the array index you need